packages/dds/tree/src/feature-libraries/optional-field/optionalFieldCodecs.ts

```diff
@@ -122,25 +115,28 @@ function makeOptionalFieldCodec<TChildChange = NodeChangeset>(
 				encoded.c = [];
 				for (const [id, childChange] of change.childChanges) {
 					encoded.c.push([
-						registerIdCodec.encode(id, context),
-						childCodec.encode(childChange, context),
+						registerIdCodec.encode(id, context.baseContext),
+						context.encodeNode(childChange as NodeId),
```

Comment:
It would be nice to get rid of this cast here. I think it would be easy enough to make `FieldChangeEncodingContext` take a template type to use instead of `NodeId`.

Reply:
Resolving this thread and adding a new one in the right place.